agent/consul/state/config_entry_test.go

```diff
@@ -1250,3 +1250,36 @@ func TestStore_ReadDiscoveryChainConfigEntries_SubsetSplit(t *testing.T) {
 	require.Len(t, entrySet.Resolvers, 1)
 	require.Len(t, entrySet.Services, 1)
 }
+
+func TestStore_ValidateGatewayNamesCannotBeShared(t *testing.T) {
```

Comment:
note: you'll probably want a TODO for adding an ent-only test that you can share the same name in different namespaces